src/components/SearchBar.tsx

```diff
@@ -192,13 +215,18 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
 
         return (
             <div className="SearchBar">
+                <div className="LibraryHeader">
+                    Library
+                    <div>
+                        |{filterBtn}|{structuredBtn}|{detailedBtn}
```

Comment:
Instead of `{filterBtn}` here, you can do `createFilterButton()`, and extract the complexities out of `render` method.